crates/bevy_ecs/src/system/query.rs

```diff
@@ -292,6 +292,31 @@ where
         }
     }
 
+    /// Gets the result of a single-result query
+    ///
+    /// If the query has exactly one result, returns the result inside `Ok`
+    /// otherwise returns either `Err(QuerySingleError::NoEntities(...))`
+    /// or `Err(QuerySingleError::MultipleEntities(...))`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///  # use bevy_ecs::system::{Query, QuerySingleError};
+    ///  # use bevy_render::camera::Camera;
+    /// fn do_something_with_camera(query: Query<&Camera>) {
+    ///     match query.single() {
+    ///         Ok(camera) => {
+    ///             // do something with our camera
+    ///         }
+    ///         Err(QuerySingleError::NoEntities(_)) => {
```

Comment:
Hmm, I don't like that you have to match against the `_` there. It doesn't actually contain any useful data.

But I'm not sure this PR is the place to fix that, if it can be fixed.